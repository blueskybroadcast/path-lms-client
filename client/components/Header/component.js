import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';

import HeaderNavDropdown from './components/HeaderNavDropdown';

const Header = ({ auth }) => {
  const authButton = auth ? (
    <a href="/api/logout">Logout</a>
  ) : (
    <a href="/api/auth/google">Login</a>
  );

  return (
    <header className="account">
      <nav aria-expanded="false">
        <div className="pull-left logo">
          <div className="item" />
          <div className="item">
            <h1>
              <a className="home" title="Showroom" href="/showroom">
                <i className="icon icon-home" />
                &nbsp;Showroom
              </a>
            </h1>
          </div>
          <div className="item toggle-nav">
            <button
              aria-expanded="false"
              className="toggle-header-nav transparent"
              title="Toggle navigation menu"
              type="button"
            >
              <i className="icon icon-reorder" />
            </button>
          </div>
        </div>
        <div className="pull-left flexible">
          <HeaderNavDropdown />
          <div className="item">
            <a
              className="courses"
              title="Programs"
              href="/showroom/courses"
            >
              <i className="icon icon-course" />
              &nbsp;Programs
            </a>
          </div>
          <div className="item">
            <NavLink
              className="content-library"
              title="Video Go"
              to="/showroom/library"
            >
              <i className="icon icon-library" />
              &nbsp;Video Go
            </NavLink>
          </div>
          <div className="item">
            <a
              className="product-bundles"
              title="Product Bundles"
              href="/showroom/product_bundles"
            >
              <i className="icon icon-product-bundle" />
              &nbsp;Product Bundles
            </a>
          </div>
          <div className="item search">
            <a
              className="searches"
              title="Search"
              href="/showroom/searches"
            >
              <i className="icon icon-search" />
              <span className="search-text">Search</span>
            </a>
          </div>
        </div>
        <div className="pull-right flexible">
          <div className="item global-searching">
            <form
              className="search"
              action="/showroom/searches"
              acceptCharset="UTF-8"
              method="post"
            >
              <input name="utf8" type="hidden" defaultValue="âœ“" />
              <input type="hidden" name="authenticity_token" defaultValue="hqQ9S/zT5lejZ/Ui5q6tAlnCQ38HHA8gSh8EZ/hszsxaUMjKZ1yMKiq6/cUvRlzw6N77nu37Nas4d6scc7dT1Q==" />
              <input type="text" name="query" id="global-search-field" defaultValue="" className="global-search" placeholder="Search..." aria-label="Search" />
              <button
                aria-label="Submit search form"
                id="search-submit"
                title="Submit search form"
                type="button"
              >
                <i className="icon icon-search" />
              </button>
            </form>
          </div>
          <div className="item cart">
            <a title="Cart" href="/showroom/cart">
              <i className="icon icon-shopping-cart" />
              <span className="cart-text">&nbsp;Cart&nbsp;</span>
              <span className="filled" data-role="cart-total">0</span>
            </a>
          </div>
          <div className="item">
            <a data-role="sign-up" id="sign-up-link" href="/showroom/sign_up">
              <i className="icon icon-asterisk" />
              &nbsp;Sign Up
            </a>
          </div>
          <div className="item">
            <a data-role="login" href="/showroom/sign_in">
              <i className="icon icon-sign-in" />
              &nbsp;Sign In
            </a>
          </div>
        </div>
      </nav>
    </header>
  );
};

const mapStateToProps = ({ auth }) => ({
  auth
});

export default connect(mapStateToProps)(Header);
