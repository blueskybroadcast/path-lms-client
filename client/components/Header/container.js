import { connect } from 'react-redux';

import {
  currentUserAttributesSelector, currentAccountSlugSelector, currentAccountNameSelector
} from '../../selectors/authSelectors';

import Header from './component';

const mapStateToProps = state => ({
  currentUser: currentUserAttributesSelector(state),
  slug: currentAccountSlugSelector(state),
  accountName: currentAccountNameSelector(state)
});

export default connect(mapStateToProps)(Header);
