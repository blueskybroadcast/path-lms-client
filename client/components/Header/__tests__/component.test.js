import React from 'react';
import { shallow } from 'enzyme';
import Header from '../component';

describe('Header', () => {
  const props = {
    slug: 'testing',
    accountName: 'account name'
  };
  test('should render Header correctly if user is admin', () => {
    const wrapper = shallow(
      <Header
        {...props}
        currentUser={{
          guest: false,
          admin: true,
          firstName: 'User'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should render Header correctly if user is guest', () => {
    const wrapper = shallow(
      <Header
        {...props}
        currentUser={{
          guest: true,
          admin: false,
          firstName: 'User'
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
