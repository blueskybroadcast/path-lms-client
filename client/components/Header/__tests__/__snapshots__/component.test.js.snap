// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header correctly if user is admin 1`] = `
<header
  className="account"
>
  <nav>
    <div
      className="pull-left logo"
    >
      <div
        className="item"
      />
      <div
        className="item"
      >
        <h1>
          <a
            className="home"
            href="/testing"
          >
            <i
              className="icon icon-home"
            />
             
            account name
          </a>
        </h1>
      </div>
      <div
        className="item toggle-nav"
      >
        <button
          className="toggle-header-nav transparent"
          title="Toggle navigation menu"
          type="button"
        >
          <i
            className="icon icon-reorder"
          />
        </button>
      </div>
    </div>
    <div
      className="pull-left flexible"
    >
      <div
        className="item admin-edit-account"
      >
        <a
          className="settings"
          href="/testing/account/edit"
          title="Admin Settings"
        >
          <i
            className="icon icon-wrench"
          />
          <span
            className="admin-text"
          >
             Admin
          </span>
        </a>
      </div>
      <HeaderNavDropdown />
      <div
        className="item"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="courses"
          title="Programs"
          to="/testing/courses"
        >
          <i
            className="icon icon-course"
          />
           Courses
        </NavLink>
      </div>
      <div
        className="item"
      >
        <a
          className="content-library"
          href="/testing/library"
          title="Video Go"
        >
          <i
            className="icon icon-library"
          />
           Content Library
        </a>
      </div>
      <div
        className="item"
      >
        <a
          className="product-bundles"
          href="/testing/product_bundles"
          title="Product Bundles"
        >
          <i
            className="icon icon-product-bundle"
          />
           Product Bundles
        </a>
      </div>
      <div
        className="item search"
      >
        <a
          className="searches"
          href="/testing/searches"
          title="Search"
        >
          <i
            className="icon icon-search"
          />
          <span
            className="search-text"
          >
            Search
          </span>
        </a>
      </div>
    </div>
    <div
      className="pull-right flexible"
    >
      <div
        className="item global-searching"
      >
        <form
          className="search"
        >
          <input
            aria-label="Search"
            className="global-search"
            defaultValue=""
            id="global-search-field"
            name="query"
            placeholder="Search..."
            type="text"
          />
          <button
            aria-label="Submit search form"
            id="search-submit"
            title="Submit search form"
            type="button"
          >
            <i
              className="icon icon-search"
            />
          </button>
        </form>
      </div>
      <div
        className="item cart"
      >
        <a
          href="/testing/cart"
          title="Cart"
        >
          <i
            className="icon icon-shopping-cart"
          />
          <span
            className="cart-text"
          >
             Cart 
          </span>
          <span
            className="filled"
            data-role="cart-total"
          >
            0
          </span>
        </a>
      </div>
      <div
        className="item avatar-wrapper"
      >
        <a
          href="/testing/profile"
          title="Profile Settings"
        >
          <i
            className="icon icon-user"
          />
          <span
            className="profile-text"
          >
             
            User
          </span>
        </a>
      </div>
      <div
        className="item sign-out-wrapper"
      >
        <a
          href="/testing/sign_out"
          title="Sign Out"
        >
          <i
            className="icon icon-sign-out"
          />
          <span
            className="sign-out-text"
          >
             Sign Out
          </span>
        </a>
      </div>
    </div>
  </nav>
</header>
`;

exports[`Header should render Header correctly if user is guest 1`] = `
<header
  className="account"
>
  <nav>
    <div
      className="pull-left logo"
    >
      <div
        className="item"
      />
      <div
        className="item"
      >
        <h1>
          <a
            className="home"
            href="/testing"
          >
            <i
              className="icon icon-home"
            />
             
            account name
          </a>
        </h1>
      </div>
      <div
        className="item toggle-nav"
      >
        <button
          className="toggle-header-nav transparent"
          title="Toggle navigation menu"
          type="button"
        >
          <i
            className="icon icon-reorder"
          />
        </button>
      </div>
    </div>
    <div
      className="pull-left flexible"
    >
      <HeaderNavDropdown />
      <div
        className="item"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="courses"
          title="Programs"
          to="/testing/courses"
        >
          <i
            className="icon icon-course"
          />
           Courses
        </NavLink>
      </div>
      <div
        className="item"
      >
        <a
          className="content-library"
          href="/testing/library"
          title="Video Go"
        >
          <i
            className="icon icon-library"
          />
           Content Library
        </a>
      </div>
      <div
        className="item"
      >
        <a
          className="product-bundles"
          href="/testing/product_bundles"
          title="Product Bundles"
        >
          <i
            className="icon icon-product-bundle"
          />
           Product Bundles
        </a>
      </div>
      <div
        className="item search"
      >
        <a
          className="searches"
          href="/testing/searches"
          title="Search"
        >
          <i
            className="icon icon-search"
          />
          <span
            className="search-text"
          >
            Search
          </span>
        </a>
      </div>
    </div>
    <div
      className="pull-right flexible"
    >
      <div
        className="item global-searching"
      >
        <form
          className="search"
        >
          <input
            aria-label="Search"
            className="global-search"
            defaultValue=""
            id="global-search-field"
            name="query"
            placeholder="Search..."
            type="text"
          />
          <button
            aria-label="Submit search form"
            id="search-submit"
            title="Submit search form"
            type="button"
          >
            <i
              className="icon icon-search"
            />
          </button>
        </form>
      </div>
      <div
        className="item cart"
      >
        <a
          href="/testing/cart"
          title="Cart"
        >
          <i
            className="icon icon-shopping-cart"
          />
          <span
            className="cart-text"
          >
             Cart 
          </span>
          <span
            className="filled"
            data-role="cart-total"
          >
            0
          </span>
        </a>
      </div>
      <div
        className="item"
      >
        <a
          href="/testing/sign_up"
        >
          <i
            className="icon icon-asterisk"
          />
           Sign Up
        </a>
      </div>
      <div
        className="item"
      >
        <a
          data-role="login"
          href="/testing/sign_in"
        >
          <i
            className="icon icon-sign-in"
          />
           Sign In
        </a>
      </div>
    </div>
  </nav>
</header>
`;
